pkg/poller/poller.go

```diff
@@ -283,10 +283,7 @@ func (c *pollingController) pollPullRequestPushHook(ctx context.Context, l *logr
 }
 
 func (c *pollingController) hasStatusForSHA(ctx context.Context, l *logrus.Entry, fullName string, sha string, isRelease bool) (bool, error) {
-	opts := scm.ListOptions{
-		Page: 1,
-	}
-	statuses, _, err := c.scmClient.Repositories.ListStatus(ctx, fullName, sha, opts)
+	statuses, err := c.ListAllStatuses(ctx, fullName, sha)
```

Comment:
can we move this function to go-scm instead?

Reply:
@ankitm123 I've tried to introduce a PR related to this to go-scm but it doesn't seem to be as active as this repository: https://github.com/jenkins-x/go-scm/pull/321

Either way, my understanding was that go-scm is providing a library that abstracts away from the APIs offered by different SCM providers, but here we are combining multiple API calls to get more functionality. Happy to close this PR though and move the logic into go-scm (although I believe this would require adding a new function to every supported provider), but if you agree with the above I can add the test that @tomhobson asked for

Reply:
I've created an initially PR: https://github.com/jenkins-x/go-scm/pull/323

If this looks like what you were thinking then I can add tests and close out this PR

Reply:
I am back from vacation, and will review this PR tonight. Sorry for the delay in getting to this PR.

Reply:
So, what I was thinking is to define another function `c.scmClient.Repositories.ListStatusAll(ctx, fullName, sha, opts)` in go-scm (checked ur go-scm repo). Instead of defining it in the client file, define it for each driver. I know some scm providers have different pagination limits (I recall having that issue with bitbucket) - that way we can easily extend the tests in go-scm.

But giving it a bit more thought:
I am fine with it being here, as u mentioned before, this is aggregation, and should be ok outside of go-scm. You can keep it here, and add the tests, and that will be ok for me.

Reply:
>but it was implemented in exactly the same way for each provider, so tried to just create a general function.

The only concern I had was with pagination size, but I think it should be ok.

Cool, looking forward to the tests yay

Reply:
I think the pagination stuff gets handled by the providers as far as I know like this: https://github.com/jenkins-x/go-scm/blob/main/scm/driver/bitbucket/util.go#L35-L37

Have added a test, but it's not the best because the fake provider doesn't consider the ListOptions and always returns all the statuses, but I think it's still okay (or not sure how to do better without changes to go-scm): https://github.com/jenkins-x/go-scm/blob/main/scm/driver/fake/repo.go#L145-L150

Reply:
Sorry for linking across repositories but this PR is also quite important for us as otherwise Lighthouse poller has a race condition by potentially hiding previous statuses with newer ones: https://github.com/jenkins-x/go-scm/pull/321

Reply:
> Have added a test, but it's not the best because the fake provider doesn't consider the ListOptions

That should be ok, we can update the fake provider, and update the tests in a subsequent PR...